plugins/services/src/js/reducers/serviceForm/FormReducers/Constraints.js

```diff
@@ -13,13 +13,22 @@ function processTransaction(state, { type, path, value }) {
   let newState = state.slice();
 
   if (type === ADD_ITEM) {
-    newState.push({ fieldName: null, operator: null, value: null });
+    let defaultValue = { fieldName: null, operator: null, value: null };
+
+    if (value != null && value.type != null) {
```

Comment:
How about `value.type === "default"` to be fully explicit in the linkage between the code below and this check?

Reply:
If we would go with `value.type === "default"` We would exclude all the other types. With the condition above we are making sure that value is not null and the type is set.

Reply:
The confusion here is that we do have a defaultValue and a default type which has nothing to do with each other. The defaultValue is the defaultValue for a constraint, the type describes the type in the future we might add types like hostname for example.